src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/bindings/BindingManager.h>
+#include <app/util/binding-table.h>
+
+namespace chip {
+
+BindingManager BindingManager::sBindingManager;
+
+CHIP_ERROR BindingManager::CreateBinding(FabricIndex fabric, NodeId node, EndpointId localEndpoint, ClusterId cluster)
+{
+    ReturnErrorCodeIf(mAppServer == nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    return EnqueueClusterAndConnect(fabric, node, localEndpoint, cluster);
+}
+
+CHIP_ERROR BindingManager::EnqueueClusterAndConnect(FabricIndex fabric, NodeId node, EndpointId endpoint, ClusterId cluster)
+{
+    FabricInfo * fabricInfo = mAppServer->GetFabricTable().FindFabricWithIndex(fabric);
```

Comment:
Please validate if we get a valid fabric info - check for null before deferencing below.